runtime/near-vm-runner/src/logic/logic.rs

```diff
@@ -2017,17 +2017,25 @@ impl<'a> VMLogic<'a> {
     }
 
     /// Creates a promise that will execute a method on the current account with given arguments
-    /// and gas. The created promise will have a special data dependency which can be submitted
-    /// using `promise_yield_resume` within `yield_num_blocks`.
+    /// and gas.
+    ///
+    /// The created promise will have a special data dependency within a certain protocol-defined
+    /// number of blocks (as defined by the `yield_timeout_length_in_blocks` parameter.)
+    ///
+    /// In order to satisfy the data dependency `promise_yield_resume` must be called with the
```

Comment:
I think "must be called" sounds scary and obscures the fact that the user is free to never call resume.